File: src/Crawler/Info.php
<?php

/*
 * This file is part of the SiteOne Crawler.
 *
 * (c) Ján Regeš <jan.reges@siteone.cz>
 */

declare(strict_types=1);

namespace Crawler;

class Info
{

    public readonly string $name;
    public readonly string $version;
    public readonly string $executedAt;
    public readonly string $command;
    public readonly string $hostname;
    public string $finalUserAgent;

    /**
     * @param string $name
     * @param string $version
     * @param string $executedAt
     * @param string $command
     * @param string $hostname
     * @param string $finalUserAgent
     */
    public function __construct(string $name, string $version, string $executedAt, string $command, string $hostname, string $finalUserAgent)
    {
        $this->name = $name;
        $this->version = $version;
        $this->executedAt = $executedAt;
        $this->command = $command;
        $this->hostname = $hostname;
        $this->finalUserAgent = $finalUserAgent;
    }

    public function setFinalUserAgent(string $finalUserAgent): void
    {
        $this->finalUserAgent = $finalUserAgent;
    }

}

File: src/info.rs
// SiteOne Crawler - Info
// (c) Jan Reges <jan.reges@siteone.cz>

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Info {
    pub name: String,
    pub version: String,
    pub executed_at: String,
    pub command: String,
    pub hostname: String,
    pub final_user_agent: String,
    /// The initial URL passed via --url option
    pub initial_url: String,
}

impl Info {
    pub fn new(
        name: String,
        version: String,
        executed_at: String,
        command: String,
        hostname: String,
        final_user_agent: String,
        initial_url: String,
    ) -> Self {
        Self {
            name,
            version,
            executed_at,
            command,
            hostname,
            final_user_agent,
            initial_url,
        }
    }

    pub fn set_final_user_agent(&mut self, final_user_agent: String) {
        self.final_user_agent = final_user_agent;
    }
}
